### PR TITLE
Fix path dots expansion

### LINKED ISSUE #3398: Can't access directores with "..." in their name even when using autocomplete
**Describe the bug**
A directory with "..." in its name does for all intents and purposes not exist to nushell. Such a directory can't be created  either as `mkdir "a...b"` results in a directory and a subdirectory in it: `a../..b`.

**To Reproduce**
1. Have a directory named say, "a...b"
2. Verify that it's there with `ls`
3. Try to cd into it in nushell by either typing the name manually or autocompleting it from the first letter(s)
4. "directory not found"

**Expected behavior**
Autocompletion should escape directory names correctly where needed.

**Configuration (please complete the following information):**

```
~
❯ version | pivot
───┬────────────────────┬────────────────────────────────────────────────────────────
 # │      Column0       │                          Column1
───┼────────────────────┼────────────────────────────────────────────────────────────
 0 │ version            │ 0.30.0
 1 │ build_os           │ linux-x86_64
 2 │ rust_version       │ rustc 1.51.0
 3 │ cargo_version      │ cargo 1.51.0
 4 │ pkg_version        │ 0.30.0
 5 │ build_time         │ 2021-04-25 10:06:26
 6 │ build_rust_channel │ release
 7 │ features           │ ctrlc, default, directories, dirs, ichwh, rustyline, term,
   │                    │ which, zip
───┴────────────────────┴────────────────────────────────────────────────────────────
```

### PR DESCRIPTION
**Pull request by kubouch**
Previously, the parser would expand `a...b` as `a../..b`. Now, >2 dots are only expanded when the whole path component consists of dots (i.e., `...` expands to `../..` while `a...b` stays as it is).

Added a couple more tests to cover this better.

Also, the OS main separator is respected (i.e., `...` expands to `../..` on Unix and `..\..` on Windows).

Fixes #3398

### CODE REVIEW

File: crates/nu-parser/src/path.rs [hunk 1]
```diff
@@ -70,21 +95,81 @@ pub fn expand_path<'a>(path: &'a str) -> Cow<'a, str> {
...
+        assert_eq!(
+            r"a...b/./c..d/../e.f/..\..\..//.",
+            &expand_ndots("a...b/./c..d/../e.f/....//.").to_string()
+        );
+    }
+
+    #[cfg(windows)]
+    #[test]
+    fn string_with_three_ndots_and_final_slash() {
+        assert_eq!(r"..\../", &expand_ndots(".../").to_string());
```
Comment (jonathandturner) [inline crates/nu-parser/src/path.rs:138]: For the final slashes on Windows, should these go in the direction that's native to the platform?

Comment (kubouch) [reply crates/nu-parser/src/path.rs:138]: Currently, what the `expand_ndots()` does is replace manydots (`....`) with a series of `..` (`../..`). Everything else is noop. So the final slash is out of scope of the code, it simply copy-pastes whatever the input is.

**Comment by jonathandturner** [comment]
Thanks!

File: crates/nu-parser/src/path.rs (hunk 1 above)
Comment (kubouch) [reply crates/nu-parser/src/path.rs:138]: At this point, the path is just a &str, so it doesn't know about filenames yet. The `expand_ndots()` just replaces any `...` with `../..` and that's all it does. If the user put `...\` as an input, the result would be `..\..\`.

Comment (kubouch) [reply crates/nu-parser/src/path.rs:138]: I'm not exactly sure what you're asking here.

**Pull request merged**